app/controllers/admin/user_orders_controller.rb

```diff
@@ -4,4 +4,15 @@ def show
     @order = Order.find(params[:order_id])
   end
 
+  def update
+    order = Order.where(id: params[:order_id], user_id: params[:user_id]).first
+    order.update(status: 'Cancelled')
+    order.item_orders.each do |item_order|
+      item_order.return_inventory if item_order.status == 'Fulfilled'
+      item_order.update_attributes(status: 'Unfulfilled')
+    end
+    flash[:notice] = 'Order has been cancelled!'
```

Comment:
This flashes a message for the admin that the order has been cancelled, but the user story didn't say anything about sending a message to the user when they log in again to notify them their order has been cancelled. I would also think that would be through a message functionality that we don't have. Thoughts?